server/src/handler.ts

```diff
@@ -153,8 +153,19 @@ export class Handler {
     return new abaplint.LanguageServer(this.reg).gotoDefinition(params);
   }
 
-  public onReferences(params: LServer.TextDocumentPositionParams): LServer.Location[] {
-    return new abaplint.LanguageServer(this.reg).references(params);
+  public onReferences(params: LServer.TextDocumentPositionParams): LServer.Location[] | undefined {
+    const server = new abaplint.LanguageServer(this.reg);
+    const references = server.references(params);
+    if (!references.length) {
```

Comment:
for my mind, its easier to understand this, instead of having to think about the type

```suggestion
    if (references.length === 0) {
```